base/threadingconstructs.jl

```diff
@@ -43,7 +43,7 @@ function _threadsfor(iter,lbody)
                 len, rem = 1, 0
             end
             # compute this thread's iterations
-            f = 1 + ((tid-1) * len)
+            f = firstindex(r) + ((tid-1) * len)
```

Comment:
I feel like I'm missing something and there should be an abstraction to represent this computation. It's something like:

```julia
# Construct a range which starts at 1 and yields the indices of `r` when indexed:
# Why don't we have a name for this??
inds = firstindex(r):lastindex(r)

f = inds[(tid-1) * len]
l = inds[tid*len - 1]
```

@andyferris I feel like you might have thought about this. Do you have a name for it?